### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by BytingMicrochips**
-add getComments functionality
-add getComments tests
-add comments-controller
-add comments-model
-updated endpoints.json

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -16,51 +16,145 @@ beforeEach(() => {
...
+        .then(({ body }) => {
+          expect(body[1]).toMatchObject(expectedShape)
+        })
+    });
+    test("Should serve comments in order of most recent first", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .then(({ body }) => {
+          expect(body[0].comment_id).toBe(5);
+          expect(body[10].comment_id).toBe(9);
```
Comment (2202Hannah) [inline __tests__/app.test.js]: There is a great jest matcher called toBeSotedBy that can be installed for this test. It makes testing for sorted by ascending or descending a little clearer in your code. Here is an example
.toBeSortedBy('created_at', {descending: true})

File: models/comments-model.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                        return rows;
+                    } else {
+                      return Promise.reject({
+                        status: 404,
+                        msg: `We couldn't find any comments on that...`,
+                      });   
+                  }
+              });
+    } else {
+        return Promise.reject({status : 400, msg: `${article_id} is not a valid article_id`})
```
Comment (2202Hannah) [inline models/comments-model.js]: Really nice use of string literals here to make your error message dynamic

File: __tests__/app.test.js [hunk 3]
```diff
@@ -16,51 +16,145 @@ beforeEach(() => {
...
+        .then(({ body }) => {
+          expect(body.msg).toBe(`burritoSupreme is not a valid article_id`);
+        })
+    });
+    test("Should send code 404 if article_id has no comments", () => {
+      return request(app)
+        .get("/api/articles/8/comments")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe(`We couldn't find any comments on that...`);
```
Comment (2202Hannah) [inline __tests__/app.test.js]: Please reconsider whether this scenario should be an error. Some articles may not have comments yet and we would subsequently expect the comments array to be empty, but this wouldn't be an error as this is behaviour we want to see.

File: controllers/comments-controller.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+//REQUIRES
+const { selectComments } = require("../models/comments-model");
+
+//FUNCTION
+function getComments(req, res, next) {
+  selectComments(req.params.article_id)
+      .then((comments) => {
+      res.status(200).send(comments);
```
Comment (2202Hannah) [inline controllers/comments-controller.js]: Please make sure you are retuning the data on an appropriate key e.g. {comments: comments}

File: models/comments-model.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+//REQUIRES
+const db = require("../db/connection.js");
+
+//FUNCTION
+function selectComments(article_id) {
+    const onlyDigitsRegex = /(^[0-9]+$)/
+    if (onlyDigitsRegex.test(article_id)) {
+            return db
+              .query("SELECT * FROM comments WHERE comments.article_id = $1 ORDER BY created_at DESC;", [article_id])
+                .then(({ rows }) => {
```
Comment (2202Hannah) [inline models/comments-model.js]: You will need to consider that there are 2 reasons that rows could be empty:
1) article exists but had no comments (status 200)
2) article does not exist (status 404)

**Review by 2202Hannah — CHANGES_REQUESTED** [review]
Nice work on ticket 6 so far. I have made some comments for you to consider. Please update your code and then make another pull request.

File: __tests__/app.test.js (hunk 1 above)
Comment (BytingMicrochips) [reply __tests__/app.test.js]: I gave up trying to get the jest docs to load yesterday (bad internet after powercut)! I had all the syntax right but hadn't put the  extra line inside the package.json so nothing was happening - got it sorted now thanks :)

**Review by Keb84 — APPROVED** [review]
Really nice work on those changes from your feedback 🚀 . Merge when ready 🎸

**Pull request merged**